scripts/lib/check_rabbitmq_queue.py

```diff
@@ -69,11 +69,13 @@ def analyze_queue_stats(queue_name: str, stats: Dict[str, Any],
         # Queue isn't updating the stats file and has some events in
         # the backlog, it's likely stuck.
         #
-        # TODO: There's an unfortunate race where if the queue has
-        # been empty for the last hour (because there haven't been 50
-        # new events in the last hour), and then gets a burst, this
-        # condition will be true for the first (event_handling_time *
-        # 50).
+        # TODO: There's an unlikely race condition here - if the queue
+        # was fully emptied and was idle due to no new events coming for over 180 seconds, suddenly
+        # gets a burst of events and this code runs exactly in the very small
+        # time window between those events popping up and the queue beginning
+        # to process the first one (which will refresh the stats file at the very start),
+        # we'll incorrectly return the CRITICAL status. The chance of that happening
+        # should be neglibile though.
```

Comment:
I'm not sure this is negligible, exactly; this condition should last TIME_PER_EVENT which could be like 1s every time that a bursty queue adds events (most commonly, when digest emails are scheduled).  We can see if it happens in practice.

Reply:
Hmm, the race is much rarer than that because we're adding a snippet in do_consume:
```
    def do_consume(self, consume_func: Callable[[List[Dict[str, Any]]], None],
                   events: List[Dict[str, Any]]) -> None:
        consume_time_seconds: Optional[float] = None
        try:
            if self.idle:
                # We're reactivating after having gone idle due to emptying the queue.
                # We should update the stats file to keep it fresh and to make it clear
                # that the queue started processing, in case the event we're about to process
                # makes us freeze.
                self.idle = False
                self.update_statistics(self.get_remaining_queue_size())
```

so the time  interval where we get the false alert if the nagios queue check runs is `(moment of adding more than 10 events to the idle queue, moment of the queue starting its do_consume]` which should be very tiny. It doesn't include the time of processing an event.